huaweicloud/resource_huaweicloud_rds_read_replica_instance_test.go

```diff
@@ -38,122 +52,68 @@ func TestAccRdsReadReplicaInstance_basic(t *testing.T) {
+	var replica instances.RdsInstanceResponse
+	name := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+	resourceLabel := "huaweicloud_rds_read_replica_instance"
+	resourceName := "huaweicloud_rds_read_replica_instance.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheckEpsID(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckRdsInstanceV3Destroy(resourceLabel),
+		Steps: []resource.TestStep{
 			{
-				Config: testAccReadRdsReplicaInstanceUpdate(nameSuffix),
+				Config: testAccReadRdsReplicaInstance_withEpsId(name),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckRdsReplicaInstanceV3Exists(resourceName, &replica),
-					// check modification of flavor
-					resource.TestCheckResourceAttr(resourceName, "flavor", "rds.pg.c2.large.rr"),
-					// check modification of volume
-					resource.TestCheckResourceAttr(resourceName, "volume.0.type", "ULTRAHIGH"),
-					resource.TestCheckResourceAttr(resourceName, "volume.0.size", "120"),
-					// check modification of tags
-					resource.TestCheckResourceAttr(resourceName, "tags.key", "value1"),
-					resource.TestCheckResourceAttr(resourceName, "tags.foo", "bar2"),
+					testAccCheckRdsInstanceV3Exists(resourceName, &replica),
+					resource.TestCheckResourceAttr(resourceName, "enterprise_project_id", HW_ENTERPRISE_PROJECT_ID_TEST),
 				),
 			},
 		},
 	})
 }
 
-func testAccCheckRdsReplicaInstanceV3Destroy(s *terraform.State) error {
-	config := testAccProvider.Meta().(*Config)
-	client, err := config.RdsV3Client(HW_REGION_NAME)
-	if err != nil {
-		return fmt.Errorf("Error creating huaweicloud rds client: %s", err)
-	}
-
-	for _, rs := range s.RootModule().Resources {
-		if rs.Type != "huaweicloud_rds_read_replica_instance" {
-			continue
-		}
-
-		id := rs.Primary.ID
-		instance, err := getRdsInstanceByID(client, id)
-		if err != nil {
-			return err
-		}
-		if instance.Id != "" {
-			return fmt.Errorf("huaweicloud_rds_read_replica_instance (%s) still exists", id)
-		}
-	}
-
-	return nil
-}
-
-func testAccCheckRdsReplicaInstanceV3Exists(n string, instance *instances.RdsInstanceResponse) resource.TestCheckFunc {
-	return func(s *terraform.State) error {
-		rs, ok := s.RootModule().Resources[n]
-		if !ok {
-			return fmt.Errorf("Not found: %s. ", n)
-		}
-
-		id := rs.Primary.ID
-		if id == "" {
-			return fmt.Errorf("No ID is set. ")
-		}
-
-		config := testAccProvider.Meta().(*Config)
-		client, err := config.RdsV3Client(HW_REGION_NAME)
-		if err != nil {
-			return fmt.Errorf("Error creating huaweicloud rds client: %s", err)
-		}
-
-		found, err := getRdsInstanceByID(client, id)
-		if err != nil {
-			return fmt.Errorf("Error checking %s exist, err=%s", n, err)
-		}
-		if found.Id == "" {
-			return fmt.Errorf("resource %s does not exist", n)
-		}
-
-		instance = found
-		return nil
-	}
-}
-
-func testAccReadRdsReplicaInstanceBasic(val string) string {
+func testAccReadRdsReplicaInstanceV3_base(val string) string {
```

Comment:
suggest use more meaningful variable name

Reply:
`testAccReadRdsReplicaInstanceV3_base` is used as the basis of configuration of `resource_huaweicloud_rds_read_replica_instance_test.go`, and its role and naming are consistent with the rules of `testAccRdsInstanceV3_base` in `resource_huaweicloud_rds_instance_v3_test.go`.
So, I think it's fine.

Reply:
`testAccReadRdsReplicaInstanceV3_base` is the configuration basis of `testAccReadRdsReplicaInstanceV3_basic` and `testAccReadRdsReplicaInstanceV3_update`